public/script.js

```diff
@@ -0,0 +1,55 @@
+//access elements from html
+const inputSearch = document.querySelector('.input-search')
+const buttSearch = document.querySelector('.btn-search')
+buttSearch.addEventListener('click', () => {
+    const data = inputSearch.value;
+    const options = {
+        method: 'POST',
+        body: JSON.stringify({ name: data }),
+        headers: { 'Content-Type': 'application/json' }
+    }
+    fetch(`/search`, options)
+    .then(res => res.json())
+    .then(data => createCards(data))
```

Comment:
what if the fetch fails then `createCards` will not receive the expected value for `data` and will cause some error, so you should handle such cases by displaying a nice message for the user ex `no result found` `try another search term`

**NOTE** this thing about how the fetch api deals with request/response errors
```js
fetch(url).then((response) => {
  // Always gets a response, unless there is network error
  // It never throws an error for 4xx or 5xx response 😟
 // so here we need to check for the response status and display the result if it is 2xx or display error message if it is 4xx or 5xx
}).catch((error) => {
  // Only network error comes here
});
```